addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/core/LxWsStateUpdateEvent.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.loxone.core;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/**
+ * An event received from Loxone Miniserver with control's state update
+ *
+ * @author Pawel Pieczul - initial commit
+ *
+ */
+class LxWsStateUpdateEvent {
+    private LxUuid uuid;
+    @SuppressWarnings("unused")
+    private LxUuid iconUuid = null;
+    private double value = -1;
+    private String text = null;
+    private int size = 0;
+
+    /**
+     * Create new state update event from binary message
+     *
+     * @param isValueEvent
+     *            true if this event updates double value, false if it updates text message
+     * @param data
+     *            buffer with binary message received from Miniserver
+     * @param offset
+     *            offset in buffer where event is expected
+     */
+    LxWsStateUpdateEvent(boolean isValueEvent, byte data[], int offset) throws IndexOutOfBoundsException {
+
+        if (data.length - offset < 24) {
+            throw new IndexOutOfBoundsException();
```

Comment:
You are right, these checks are redundant, array access and ByteWrap will throw this exception for me.